shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -508,30 +509,50 @@ private int guessBottomKeyboardInset(WindowInsets insets) {
   public final WindowInsets onApplyWindowInsets(@NonNull WindowInsets insets) {
     WindowInsets newInsets = super.onApplyWindowInsets(insets);
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+      Insets systemGestureInsets = insets.getSystemGestureInsets();
```

Comment:
We don't have to do this now, but this got deprecated after 1 Android version 🤣. We could add an R check for the new API.

Reply:
Oh you're already doing getInsets below. If that works in Q, we should just merge it.

Reply:
Unfortunately, getInsets is API 30 only, so to support this for Q, we still need to do this. For R and above though, we can move to getInsets

Reply:
Yup, SG